orttraining/orttraining/training_api/utils.h

```diff
@@ -10,6 +10,42 @@
 namespace onnxruntime {
 namespace training {
 namespace api {
+
+struct ModelIdentifiers {
+  // ModelIdentifiers struct enables an easy way to store and identify the models used for training, evaluation
+  // and model updates(optimizer model).
+  // The model can be specified by a path to the model file or by a span of bytes containing the model data.
+  // Training model is required, evaluation and optimizer models are optional.
+  std::variant<std::string, gsl::span<const uint8_t>> train_model;
+  std::variant<std::optional<std::string>, gsl::span<const uint8_t>> eval_model;
+  std::variant<std::optional<std::string>, gsl::span<const uint8_t>> optim_model;
+
+  ModelIdentifiers(std::variant<std::string, gsl::span<const uint8_t>> training_model,
+                   std::variant<std::optional<std::string>, gsl::span<const uint8_t>> evaluation_model,
+                   std::variant<std::optional<std::string>, gsl::span<const uint8_t>> optimzer_model)
+      : train_model(training_model), eval_model(evaluation_model), optim_model(optimzer_model) {}
+
+  bool IsOptimizerModelAvailable() const {
+    if ((std::holds_alternative<std::optional<std::string>>(optim_model) &&
+         std::get<std::optional<std::string>>(optim_model).has_value()) ||
+        (std::holds_alternative<gsl::span<const uint8_t>>(optim_model) &&
+         std::get<gsl::span<const uint8_t>>(optim_model).size() > 0)) {
+      return true;
+    }
+    return false;
+  }
+
+  bool IsEvalModelAvailable() const {
```

Comment:
Is the logic for `IsEvalModelAvailable` and `IsOptimizerModelAvailable` the same except for the variable name?

Shall we abstract it into a common private method `IsModelAvailable`?

We can get to this later as well (in a future PR).